vllm/model_executor/layers/linear.py

```diff
@@ -97,6 +98,21 @@ def apply(self,
         Expects create_weights to have been called before on the layer."""
         raise NotImplementedError
 
+    def create_weights_moe(self, layer: torch.nn.Module, num_experts: int,
+                           hidden_size: int, intermediate_size: int,
+                           params_dtype: torch.dtype, **extra_weight_attrs):
+        raise NotImplementedError(
+            "Moe models are not yet supported for your LinearMethod")
+
+    def apply_moe(self,
+                  layer: torch.nn.Module,
+                  x: torch.Tensor,
+                  router_logits: torch.Tensor,
+                  top_k: int,
+                  renormalize: bool = True) -> torch.Tensor:
+        raise NotImplementedError(
+            "Moe models are not yet supported for your LinearMethod")
```

Comment:
This API is a bit weird. Given that our MoE actually has nothing to do with Linear layers, it might be better to have a separate `moe.py` with `class MoE`, so that we can keep unified APIs `create_weights` and `apply`.

Reply:
Sounds good - I almost did it that way. Will restructure